Include/Rover/Box.hpp

```diff
@@ -10,13 +12,91 @@ namespace Rover {
     \tparam T The type of value to generate.
   */
   template<typename T>
-  class Box : private Noncopyable {
+  class Box {
```

Comment:
```Box``` must disable copy-construction, you don't have to use ```Noncopyable``` but you will need to find a way to disable the copy constructor and assignment operator.

Reply:
As we currently intend to use Box as a dependency handling mechanism, disabling copying for it would make it impossible to copy any generator (with sub-generators). It is not something we want, is it?

Reply:
That's the same reason shared_ptr had to be used.

Reply:
```Box``` is not itself the full blown solution to dependency handling, it is a small part of the solution. What ```Box``` does do is implement type-erasure. For dependency handling there will be another concept introduced that performs "autoboxing". These terms come from languages like Haskell/Java and the functional programming world where the compiler can automatically perform type erasure and boxing/autoboxing. Unfortunately C++ is not capable of this so we have to programming that functionality manually piece by piece.

Reply:
Do you mean Generators, for example Range, are not supposed to hold Boxes as data members? At what point should we implement the autoboxing then?

Reply:
Copy & copy assignment by default with unique_ptr, but Noncopyable makes the code and error messages clearer. Let me know if you want me to remove it.

Reply:
Dependency tests are also removed as storing Boxes as data members is no longer a valid option.